mergoo/models/modeling_llama.py

```diff
@@ -541,6 +541,26 @@ def _flash_attention_forward(
             cu_seqlens_q, cu_seqlens_k = cu_seq_lens
             max_seqlen_in_batch_q, max_seqlen_in_batch_k = max_seq_lens
 
+            input_dtype = query_states.dtype
+            if input_dtype == torch.float32:
```

Comment:
You might need to add a check to ensure that query_states, key_states, and value_states all have the same dtype. This would  avoid this snippet when training/inference is in float32, and all the dtypes are as expected.

Reply:
Sure, no problem :)